src/presign/participant.rs

```diff
@@ -305,6 +270,52 @@ impl PresignParticipant {
         Ok(vec![message])
     }
 
+    fn handle_round_two_msg<R: RngCore + CryptoRng>(
+        &mut self,
+        rng: &mut R,
+        message: &Message,
+        main_storage: &Storage,
+    ) -> Result<Vec<Message>> {
+        let (auxinfo_identifier, keyshare_identifier) =
+            self.get_associated_identifiers_for_presign(&message.id())?;
+
+        // First, verify the bytes of the round two value, and then
+        // store it locally. In order to v
+        self.validate_and_store_round_two_public(
+            main_storage,
+            message,
+            auxinfo_identifier,
+            keyshare_identifier,
+        )?;
+
+        // Since we are in round 2, it should certainly be the case that all
+        // public auxinfo for other participants have been stored, since
+        // this was a requirement to proceed for round 1.
+        assert!(has_collected_all_of_others(
+            &self.other_participant_ids,
+            main_storage,
+            StorableType::AuxInfoPublic,
+            auxinfo_identifier
+        )?);
```

Comment:
I see the code could possibly panic in several places due to `assert!`s. In general I'm hesitant to introduce more panics, specially if we eventually plan to call into this library.

I suggest returning an error instead. Either from `has_collected_all_of_others` or checking this condition here and generating the error here. Whichever is appropriate.

Reply:
Adding this to issue #13